### PR TITLE
SecurityRegistry API Clean-up

### PR DESCRIPTION
**Pull request by sbernard31**
As explain in #144, we decide to not expose `SecurityRegistry` anymore. We think we don't need all the `add`, `remove` methods. **The Leshan contract is now limited to the** `SecurityStore` **interface**.

All RPK and certificates server stuff was removed from `SecurityRegistry` too.

I create a new **internal interface** `EditableSecurityStore` which can be used to easily switch between SecurityStore implementation in LeshanServerDemo and for integration tests.

### CODE REVIEW

File: leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/SecureIntegrationTestHelper.java [hunk 1]
```diff
@@ -240,7 +240,7 @@ public void createServerWithRPK() {
         builder.setLocalSecureAddress(new InetSocketAddress(InetAddress.getLoopbackAddress(), 0));
         builder.setPublicKey(serverPublicKey);
         builder.setPrivateKey(serverPrivateKey);
-        builder.setSecurityStore(new SecurityRegistryImpl() {
+        builder.setSecurityStore(new FileSecurityStore() {
             // TODO we should separate SecurityRegistryImpl in 2 registries :
```
Comment (msangoi) [inline leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/SecureIntegrationTestHelper.java]: I guess you can now remove this comment :)

File: leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/SecureIntegrationTestHelper.java [hunk 2]
```diff
@@ -265,7 +265,7 @@ public void createServerWithX509Cert(Certificate[] trustedCertificates) {
         builder.setCertificateChain(serverX509CertChain);
         builder.setTrustedCertificates(trustedCertificates);
         builder.setSecurityStore(
-                new SecurityRegistryImpl() {
+                new FileSecurityStore() {
                     // TODO we should separate SecurityRegistryImpl in 2 registries :
```
Comment (msangoi) [inline leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/SecureIntegrationTestHelper.java]: Same here and in the RedisIntegrationTestHelper class.

Comment (msangoi) [inline leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/SecureIntegrationTestHelper.java]: You can ignore my comments. It's done in another PR :) (#224)

**Comment by sbernard31** [comment]
Integrated in master (commit  	1e870ea, b338532, 4fc59f1)
